sentinel/tests/unit/schedule/outbound.js

```diff
@@ -0,0 +1,522 @@
+
+    it('gets all queues and associated docs', () => {
+      const task = {
+        _id: 'task:outbound:some_doc_id',
+        doc_id: 'some_doc_id',
+        queue: ['some', 'task']
+      };
+      const doc = {
+        _id: 'some_doc_id',
+        some: 'data'
+      };
+
+      const dbSentinelAllDocs = sinon.stub(db.sentinel, 'allDocs');
+      const dbMedicAllDocs = sinon.stub(db.medic, 'allDocs');
+
+      dbSentinelAllDocs.resolves({
+        rows: [{
+          doc: task
+        }]
+      });
+
+      dbMedicAllDocs.resolves({
+        rows: [{
+          doc: doc
+        }]
+      });
+
+      lineage.resolves([doc]);
+
+      return outbound.__get__('queuedTasks')()
+        .then(results => {
+          assert.deepEqual(results, [
+            {taskDoc: task, medicDoc: doc}
+          ]);
+        });
+    });
+  });
+  describe('mapDocumentToOutbound', () => {
+    it('supports simple dest => src mapping', () => {
+      const doc = {
+        _id: 'test-doc',
+        foo: 42,
+        bar: 'baaa'
+      };
+
+      const conf = {
+        key: 'test-config',
+        mapping: {
+          'api_foo': 'doc.foo',
+          'bar': 'doc.bar'
+        }
+      };
+
+      assert.deepEqual(outbound.__get__('mapDocumentToOutbound')(doc, conf), {
+        api_foo: 42,
+        bar: 'baaa'
+      });
+    });
+    it('supports deep dest => src mapping', () => {
+      const doc = {
+        _id: 'test-doc',
+        reported_date: 42,
+        fields: {
+          foo: 'baaaaa'
+        }
+      };
+
+      const conf = {
+        key: 'test-config',
+        mapping: {
+          'when': 'doc.reported_date',
+          'data.the_foo': 'doc.fields.foo'
+        }
+      };
+
+      assert.deepEqual(outbound.__get__('mapDocumentToOutbound')(doc, conf), {
+        when: 42,
+        data: {
+          the_foo: 'baaaaa'
+        }
+      });
+    });
+    it('errors if src path does not exist', () => {
+      const doc = {
+        _id: 'test-doc',
+        reported_date: 42,
+        fields: {
+          not_foo: 'baaaaa'
+        }
+      };
+
+      const conf = {
+        key: 'test-config',
+        mapping: {
+          'foo': 'doc.fields.foo'
+        }
+      };
+
+      assert.throws(() => outbound.__get__('mapDocumentToOutbound')(doc, conf), `Mapping error for 'test-config/foo'`);
+    });
+    it('supports optional path mappings', () => {
+      const doc = {
+        _id: 'test-doc',
+        reported_date: 42,
+        fields: {
+          not_foo: 'baaaaa',
+          bar: 42
+        }
+      };
+
+      const conf = {
+        key: 'test-config',
+        mapping: {
+          'foo': {
+            path: 'doc.fields.foo',
+            optional: true
+          },
+          'bar': 'doc.fields.bar'
+        }
+      };
+
+      assert.deepEqual(outbound.__get__('mapDocumentToOutbound')(doc, conf), {
+        bar: 42
+      });
+    });
+    it('supports basic awkward data conversion via arbitrary expressions', () => {
```

Comment:
Worth to add a test here for the case where `srcValue` is falsy but `!required`?

Reply:
modified existing test to have a falsy value in it